### PR TITLE
Fix edge function compatibility

### PR DESCRIPTION
**Pull request by schehata**
Edge functions have middleware signature and should be treated same way as a middleware instead of being handled as a lambda api function.

### CODE REVIEW

File: src/withAxiom.ts [hunk 1]
```diff
@@ -122,10 +123,15 @@ function isNextConfig(param: WithAxiomParam): param is NextConfig {
...
-  return typeof param == 'function' && !!process.env.LAMBDA_TASK_ROOT;
+  const isFunction = typeof param == 'function';
+
+  // check if running locally
+  if (process.env.NODE_ENV === 'development') {
+    const isLocalWorker = caller() == 'evalmachine.<anonymous>';
+    return isFunction && !isLocalWorker;
+  }
+  const isLambda = !!process.env.LAMBDA_TASK_ROOT;
+  return isFunction && isLambda;
```
Comment (bahlo) [inline src/withAxiom.ts:134]: Maybe we can return early if it's not a function? Might make things easier to read 🤔

File: src/withAxiom.ts [hunk 2]
```diff
@@ -108,6 +108,7 @@ function withAxiomNextMiddleware(handler: NextMiddleware): NextMiddleware {
       ev.waitUntil(log.flush());
```
Comment (bahlo) [inline src/withAxiom.ts:108]: Can we rename this function `withAxiomNextMiddleware => withAxiomNextEdgeFunction`?

**Review by bahlo — APPROVED** [review]

File: src/withAxiom.ts (hunk 1 above)
Comment (schehata) [reply src/withAxiom.ts:137]: updated

File: src/withAxiom.ts [hunk 3]
```diff
@@ -101,13 +101,14 @@ function withAxiomNextApiHandler(handler: NextApiHandler): NextApiHandler {
...
 
-function withAxiomNextMiddleware(handler: NextMiddleware): NextMiddleware {
+function withAxiomNextEdgeFunction(handler: NextMiddleware): NextMiddleware {
   return async (req, ev) => {
     try {
       const res = await handler(req, ev);
       ev.waitUntil(log.flush());
       return res;
     } catch (error) {
+      console.log('Error in edge function', error);
```
Comment (bahlo) [inline src/withAxiom.ts:111]:
```suggestion
```

Not sure if we should `console.log` here, errors will appear twice with this: Unstructured (`console.log`) and structured (`log.error`).

File: src/withAxiom.ts (hunk 1 above)
Comment (bahlo) [reply src/withAxiom.ts:137]: We don't need `isFunction` in the return statements anymore, it's always true

**Pull request merged**